File: monitoring-and-billing-platform/src/main/java/com/water/monitoring_and_billing_platform/service/DashboardService.java
package com.water.monitoring_and_billing_platform.service;

import com.water.monitoring_and_billing_platform.dto.*;

public interface DashboardService {

    DashboardResponse getMainAdminDashboard();

    CommunityAdminResponse getCommunityDashboard(Long communityId);

    UserDashboardResponse getUserDashboard(Long residentId);

    ResidentDashboardResponse getResidentDashboard(String email);

    CommunityAdminDashboardResponse getCommunityAdminDashboard(String email);

    MainAdminDashboardResponse getMainAdminDashboardData();

}

File: frontend/src/services/DashboardService.js
import api from "./api";

export const getMainDashboard = () => {
    return api.get("/dashboard/main-admin");
};

export const getCommunityDashboard = (communityId) => {
    return api.get(`/dashboard/community-admin/${communityId}`);
};

export const getUserDashboard = (residentId) => {
    return api.get(`/dashboard/user/${residentId}`);
};

export const getCommunityAdminDashboard = () => {
    return api.get("/dashboard/community-admin");
};

export const getResidentDashboard = () => {
    return api.get("/dashboard/resident");
};